src/main/scala/symsim/examples/concrete/continuousmaze/ContinuousMaze.scala

```diff
@@ -0,0 +1,139 @@
+package symsim
+package examples.concrete.continuousmaze
+
+import cats.{Eq, Foldable, Monad}
+import cats.kernel.BoundedEnumerable
+
+import org.scalacheck.{Arbitrary, Gen}
+
+import symsim.concrete.Randomized
+
+/**
+ *
+ */
+
+case class MazeState (x: Double, y: Double):
+  override def toString: String = s"[x=$x, y=$y]"
+
+type MazeObservableState = MazeState
+type MazeReward = Double
+
+enum Direction:
+  case R, L, U, D
+
+enum Velocity:
+  case B, S
+
+type MazeAction = (Direction, Velocity)
+
+val MAZE_LENGTH: Int = 4
+val MAZE_WIDTH: Int = 3
+
+val BIG_STEP: Double = 1
+val SMALL_STEP: Double = 0.5
+
+object ContinuousMaze
+  extends 
+    Agent[MazeState, MazeObservableState, MazeAction, MazeReward, Randomized],
+    Episodic:
+
+  val TimeHorizon: Int = 2000
+
+  def isFinal (s: MazeState): Boolean =
+    s == MazeState (MAZE_LENGTH, MAZE_WIDTH) || s == MazeState (MAZE_LENGTH, MAZE_WIDTH - 1)
+
+  // Maze is discrete
+  def observe (s: MazeState): MazeObservableState = MazeState (s.x.floor, s.y.floor)
+
+  // We are not using the original reward function from AIAMA as it
+  // gives to unstable learning results
+  private def mazeReward (s: MazeState): MazeReward = s match
+    case MazeState (MAZE_LENGTH, MAZE_WIDTH) => +0.0     // Good final state
+    case MazeState (MAZE_LENGTH, y) if y == MAZE_WIDTH - 1 => -1000.0   // Bad final state (dead)
+    case MazeState (_, _) => -1.0
+
+
+  def distort (a: MazeAction): Randomized[MazeAction] = a match
+    case (Direction.U, a._2) | (Direction.D, a._2) =>
+      Randomized.oneOf ((Direction.R, a._2), (Direction.L, a._2))
+    case (Direction.L, a._2) | (Direction.R, a._2) =>
+      Randomized.oneOf((Direction.U, a._2), (Direction.D, a._2))
+
+
+  def successor (s: MazeState) (a: MazeAction): MazeState =
+    require (valid (s))
+
+    val result = a match
+      case (Direction.U, Velocity.B) => MazeState (s._1, s._2 + BIG_STEP)
+      case (Direction.U, Velocity.S) => MazeState (s._1, s._2 + SMALL_STEP)
+      case (Direction.D, Velocity.B) => MazeState (s._1, s._2 - BIG_STEP)
+      case (Direction.D, Velocity.S) => MazeState (s._1, s._2 - SMALL_STEP)
+      case (Direction.R, Velocity.B) => MazeState (s._1 + BIG_STEP, s._2)
+      case (Direction.R, Velocity.S) => MazeState (s._1 + SMALL_STEP, s._2)
+      case (Direction.L, Velocity.B) => MazeState (s._1 - BIG_STEP, s._2)
+      case (Direction.L, Velocity.S) => MazeState (s._1 - SMALL_STEP, s._2)
+    if valid (result) then result else s
+
+  def valid (s: MazeState): Boolean =
+     s._1 >= 1 && s._1 <= 4 && s._2 >= 1 && s._2 <= 3 && s != MazeState (2, 2)
```

Comment:
I have just learned something new about Scala 3. I had no idea that `s._1` is legal, when `s` is an instance of a case class (https://users.scala-lang.org/t/old-style-tuple-accessors-on-normal-case-classes-in-scala-3/8394). Still I think this is confusing for the reader. You have named the fields `x` and `y`, so `s.x` would be way more natural and readable.